mirage/server/dns_server_mirage.mli

```diff
@@ -2,14 +2,18 @@
 
 module Make (P : Mirage_clock.PCLOCK) (M : Mirage_clock.MCLOCK) (T : Mirage_time.S) (S : Mirage_stack.V4) : sig
 
-  val primary : ?on_update:(old:Dns_trie.t -> Dns_server.Primary.s -> unit Lwt.t) ->
+  val primary :
+    ?on_update:(old:Dns_trie.t -> [`raw] Domain_name.t option -> Ipaddr.V4.t -> Dns_server.Primary.s -> unit Lwt.t) ->
```

Comment:
Just curious about the style here -- any reason not to label all the arguments so it's clear what (e.g.) the Ipaddr.V4.t represents?  I realise that conventionally we use labels to distinguish similar types for different uses, but in this case it's not clear what some of these represent without inspecting the code more closely.  However, this is a broader comment and not a blocker to this PR, since you are just extending the existing style by adding an argument.

Reply:
agreed -- I added labels in 169dee7